### DISCUSSION TITLE
Epiverse-TRACE websites redesign

### DISCUSSION THREAD

**Discussion by Bisaloo** [epiverse-trace/epiverse-trace.github.io#133] [General]
We currently have a large number of pages on https://epiverse-trace.github.io/, built with a variety of frameworks:

- [`epiverse-trace.github.io`](https://github.com/epiverse-trace/epiverse-trace.github.io) is a quarto blog/website
- [`blueprints`](https://github.com/epiverse-trace/epiverse-trace.github.io) is a quarto book
- packages have pkgdown documentation websites (e.g., https://epiverse-trace.github.io/linelist/)
- tutorials are based on the Carpentries' workbench (e.g., https://epiverse-trace.github.io/tutorials/)

This has led to a lack of visual harmony across the different pages, and lack of clarity that they belong to the same project. The website redesign addresses this by applying consistent graphical design rules across the different sites/frameworks.

The redesign is planned to be live on Tuesday 14 November but it can already be beta-tested. *In particular, maintainers should test the impact of the redesign on the websites associated to the repositories they are responsible for.*

Please report any issues here so we can triage them between blocking & non-blocking for the deployment. In particular, please focus your feedback on how the redesign may be breaking a site and make it less usable or accessible.

Here are the instructions to preview the redesigned pages:

## Quarto book (blueprints) & Quarto website (epiverse-trace.github.io)

Switch to the `epiversetheme` branch and run `quarto preview`.

![Screen Shot 2023-10-24 at 19 00 17-fullpage](https://github.com/epiverse-trace/epiverse-trace.github.io/assets/10783929/1e690f13-be3a-41cf-9b6b-89897298697e)

## pkgdown websites

Install the `epiversetheme` R package from the `dev` branch and re-render the pkgdown website. To do so, go to your project folder and run:

```r
pak::pak("epiverse-trace/epiversetheme@dev")
pkgdown::build_site()
```

![Screen Shot 2023-10-24 at 19 14 48-fullpage](https://github.com/epiverse-trace/epiverse-trace.github.io/assets/10783929/5ec8e37b-8c32-4ffe-abaa-9a8be726e1ad)

## Carpentries' workbench lessons

Install the `varnish` R package from our fork and re-render the lesson materials. To do so, go to your project folder and run:

```r
pak::pak("epiverse-trace/varnish@epiversetheme")
sandpaper::serve()
```

![Screen Shot 2023-10-25 at 16 47 34-fullpage](https://github.com/epiverse-trace/epiverse-trace.github.io/assets/10783929/4590ef28-46f8-4419-8e62-5d81fd3c20c3)

**Comment by bahadzie** [comment]
Will this be a good time to incorporate dark mode support in our websites?

I've switched to dark mode for over a year, both online and in Ubuntu **(Settings, Appearance)** and on my mobile phone  and I find it difficult to view website that don't support it natively. 

It's one of those one way changes. Once you switch you don't know how you survived previously.

[The bslib page](https://rstudio.github.io/bslib/) now supports dark mode. Here is the [issue thread from their GH](https://github.com/r-lib/pkgdown/issues/1696) that includes a PR for implementing dark mode which we can include in {epiversetheme}

Happy to submit a PR if we are going to support it.

**Reply by Bisaloo** [reply]
Thanks for the feedback!

I would definitely be interested in this from a personal point of view but it seems like a non-trivial change to get it fully right and keep the visual identity. So let's get this change out of the door first and add the dark theme in a second step.

Couple of timing considerations:

- let's wait until it's on `main` in `r-lib/pkgdown` so they iron out all issues, and so we don't have to install from a branch that may go out of date
- we need to synchronize this change in pkgdown websites with the same change in the quarto repositories & the workbench repositories 

I'll also coordinate with David to get guidance on what an Epiverse dark theme should look like.

**Reply by pratikunterwegs** [reply]
I would agree with Banky and push back on it being a non-trivial change. The [{preferably} package](https://preferably.amirmasoudabdol.name/index.html) provides a theme that switches based on users' system-wide settings, so I would expect that it should be possible to build off this.

**Comment by joshwlambert** [comment]
I can confirm that all the packages I maintain render correctly. Nice work on the redesign!

**Comment by bahadzie** [comment]
I can confirm that _numberize_ renders correctly.

**Comment by jamesmbaazam** [comment]
I can confirm it renders correctly for {epichains} and {bpmodels}. However, there is an overlap between the last letter of the package and the colon. See screenshots. The alternatives would either to either remove the italisation but alternatively apply a bold format to the package name as in the screenshot. 
![Screenshot 2023-11-09 at 11 40 03](https://github.com/epiverse-trace/epiverse-trace.github.io/assets/43169674/81015c80-0f9d-4611-9069-b33809718554)
![Screenshot 2023-11-09 at 11 43 20](https://github.com/epiverse-trace/epiverse-trace.github.io/assets/43169674/d963df5c-34e2-4b45-a2cf-a5305327abc4)
![Screenshot 2023-11-09 at 11 49 34](https://github.com/epiverse-trace/epiverse-trace.github.io/assets/43169674/713e861a-ccf1-4b07-a7ea-4dced8c5fcdc)

**Reply by Bisaloo** [reply]
Thanks for the note. It looks like this problem was already present but is exacerbated by the next font.

Unfortunately, I cannot selectively style this element as there is no way to select it. My first reaction would probably be to recommend sticking to the tidyverse style: package names are not styled (no italics, no bold). I'll keep thinking if there is another solution.

**Comment by Juanmontenegro99** [comment]
Hi!, I can confirm it renders correctly for {epiCo}.
![image](https://github.com/epiverse-trace/epiverse-trace.github.io/assets/60274234/1bfa02a7-3f62-40d0-9b94-8353d741596c)

**Comment by pratikunterwegs** [comment]
This theme works for the packages I maintain.

I don't think the new design is good, and I would recommend sticking to the default options provided by {pkgdown} via Bootswatch. The default theme includes an option for a black nav bar and it must be possible to add a logo with a link as well. I would also recommend not specifying a font, but rather defaulting to system fonts. The dark grey colour used for the package version does not have enough contrast with the nav bar.

**Reply by Bisaloo** [reply]
Thanks for the report, the version number colour has been updated.

**Comment by avallecam** [comment]
On the workbench side, I found some interesting things to share.

First, in full size, it renders ok. I like the white letters over a dark background, this makes the names of the navigation bar much more visible due to colour contrast.

However, I noticed that the title of the lesson is omitted in the output. I refer to the one that is configured in the `config.yaml` file, [example for tutorials repo](https://github.com/epiverse-trace/tutorials/blob/f8d106084cd1a3940ef3965ad169e96ee76a2523/config.yaml#L14) and [a workshop repo](https://github.com/epiverse-trace/research-compendium/blob/8ae2bf08978bcde36eadfdd528f99f210981c9cc/config.yaml#L14). Please see figure below:

![image](https://github.com/epiverse-trace/epiverse-trace.github.io/assets/21230390/65ec180b-dd35-40ee-8221-f1102f61028a)

Also, when I moved the window to half of the screen, the format disappeared and the organization logo got lost in the white background. But then I was able to read the lesson title, "Epiverse-TRACE" in this example. The threshold appears when the width of the page makes the navigation bar hide in a navigation box in the upper left corner. See the figure below:

![image](https://github.com/epiverse-trace/epiverse-trace.github.io/assets/21230390/4f0ae401-0605-4662-9070-6df1f6362f61)

Now that the logo will have the full name, we can think of replacing the current title ([issue created](https://github.com/epiverse-trace/tutorials/issues/62)), but the organization is also hosting other lessons on different topics, so probably we need to keep it visible.

Edit: Does moving the window to half of the screen replicate how it looks on a cellphone?

**Reply by avallecam** [reply]
Minor note: While testing this half-scree behaviour with the navigation bar now as a navigation box, after opening this navigation menu:

![image](https://github.com/epiverse-trace/epiverse-trace.github.io/assets/21230390/2b14bff3-400e-4de0-bf63-e57811659325)

And then closing it with the X in the upper right corner, sometimes it gets like this, white a blank space between the top and the content far below:

![image](https://github.com/epiverse-trace/epiverse-trace.github.io/assets/21230390/468cf67c-7013-4683-ad84-ac1cd3182f60)

But this also happened with the other sites online ([web 1](https://epiverse-trace.github.io/tutorials/index.html), [web 2](https://epiverse-trace.github.io/research-compendium/)) at the same moment and under the same size conditions, so we should not worry about it. This was solved by refreshing the page. Just reporting in case you also find this.

**Reply by Bisaloo** [reply]
Thanks for the report! Having other examples of lessons definitely helped. All issues should be fixed now.

**Reply by avallecam** [reply]
Thank you, I confirm they are solved. 

One additional observation. In the movement of the window to the half of the screen, we lose the logo on top. That logo is also visible when we open the tutorial in a cellphone.

Here is the default style:

![image](https://github.com/epiverse-trace/epiverse-trace.github.io/assets/21230390/afb7ed72-afa6-4cd7-a216-5576936b8607)

Here is how it looks:

![image](https://github.com/epiverse-trace/epiverse-trace.github.io/assets/21230390/ae428e99-8ec5-4b22-9d4f-5e3731f0501e)

**Comment by GeraldineGomez** [comment]
Hi, I can confirm it renders correctly for `{sivirep}`, `{vaccineff}` , and `{serofoi}`. However, the package names appear in black, making it impossible to visualize them in the header. I've tried building the sites several times and visualizing them in different browsers.

![sivirep](https://github.com/epiverse-trace/epiverse-trace.github.io/assets/46997659/90845116-d6cf-42cf-b155-c403a72617e5)

![vaccineff](https://github.com/epiverse-trace/epiverse-trace.github.io/assets/46997659/444fd039-135f-4257-9964-7740b1edc8dd)

![serofoi](https://github.com/epiverse-trace/epiverse-trace.github.io/assets/46997659/0dd23e62-20c0-4c99-af3c-de672f1bdca7)

**Reply by ntorresd** [reply]
Hi! I confirm it renders correctly for {serofoi}. I don't seem to have the same issue as @GeraldineGomez in the name of the package though. 

![serofoi-new-template](https://github.com/epiverse-trace/epiverse-trace.github.io/assets/45337127/b6b6fdd5-c48e-4037-9d77-92220f2e3fbc)

**Reply by davidsantiagoquevedo** [reply]
Hi! Works for {vaccineff}. The name of the package is rendered correctly in my case.
![image](https://github.com/epiverse-trace/epiverse-trace.github.io/assets/67650252/c343ac48-84a6-4e19-94ed-7f67f8040f4a)

**Reply by Bisaloo** [reply]
Hi Geraldine, thanks for the report! I've noticed this issue in the screenshot @Juanmontenegro99 posted and it has been fixed yesterday. All should be fine if you update epiversetheme:

```r
pak::pak("epiverse-trace/epiversetheme@dev")
```

**Comment by Bisaloo** [comment]
The Harmonization step of the redesign is complete. Middle- & long-term suggestions from this thread, in particular the dark theme, have been written down for potential future next steps.

I'm closing this discussion thread. Please open any issues in the relevant repositories from now on.
